openmeetings-service/src/main/java/org/apache/openmeetings/service/calendar/caldav/handler/EtagsHandler.java

```diff
@@ -0,0 +1,189 @@
+
+            CalendarQuery query = new CalendarQuery(properties, vcalendar, calendarData, false, false);
+
+            CalDAVReportMethod reportMethod = null;
+
+            try {
+                reportMethod = new CalDAVReportMethod(path, query, CalDAVConstants.DEPTH_1);
+                client.executeMethod(reportMethod);
+                if(reportMethod.succeeded()){
+                    MultiStatusResponse[] multiStatusResponses = reportMethod.getResponseBodyAsMultiStatus().getResponses();
+
+                    //Parse the responses into Appointments
+                    for(MultiStatusResponse response: multiStatusResponses){
+                        if(response.getStatus()[0].getStatusCode() == DavServletResponse.SC_OK) {
+                            String etag = CalendarDataProperty.getEtagfromResponse(response);
+                            Calendar ical = CalendarDataProperty.getCalendarfromResponse(response);
+                            Appointment appointments = utils.parseCalendartoAppointment(
+                                    ical, response.getHref(), etag, calendar);
+
+                            appointmentDao.update(appointments, ownerId);
+                        }
+                    }
+                }
+                else log.error("Report Method return Status: " + reportMethod.getStatusCode()
+                        + " for calId" + calendar.getId());
+
+            } catch (IOException | DavException e) {
+                log.error("Error during the execution of calendar-multiget Report.");
+            } catch (Exception e) {
+                log.error("Severe Error during the execution of calendar-multiget Report.");
+            } finally {
+                if(reportMethod != null)
+                    reportMethod.releaseConnection();
+            }
+        }
+        else {
+            //Calendar has been inited before
+
+            DavPropertyNameSet properties = new DavPropertyNameSet();
+            properties.add(DavPropertyName.GETETAG);
+
+            CompFilter vcalendar = new CompFilter(Calendar.VCALENDAR);
+            vcalendar.addCompFilter(new CompFilter(Component.VEVENT));
+
+            CalDAVReportMethod reportMethod = null;
+
+            try {
+
+                CalendarQuery calendarQuery = new CalendarQuery(properties, vcalendar, null, false, false);
+
+                reportMethod = new
+                        CalDAVReportMethod(path, calendarQuery, CalDAVConstants.DEPTH_1);
+                client.executeMethod(reportMethod);
+                if(reportMethod.succeeded()){
+                    List<String> currenthrefs = new ArrayList<String>();
+
+                    MultiStatusResponse[] multiStatusResponses = reportMethod.getResponseBodyAsMultiStatus().getResponses();
+
+                    for(MultiStatusResponse response: multiStatusResponses){
+                        if(response.getStatus()[0].getStatusCode() == DavServletResponse.SC_OK) {
```

Comment:
Not really, each response as defined in the MultiStatusResponse as mentioned in [RFC 2518 Section 12.9.1.1](https://tools.ietf.org/html/rfc2518#section-12.9.1.1) specifies:

> The propstat XML element MUST contain one prop XML element and one status XML element.